samples/cellular/nrf_cloud_multi_service/src/cloud_connection.c

```diff
@@ -551,6 +551,31 @@ static int setup_cloud(void)
 	return 0;
 }
 
+static void wait_for_credentials(void) {
+#if defined(CONFIG_NRF_CLOUD_CHECK_CREDENTIALS)
+	LOG_INF("Waiting for necessary credentials...");
+
+	while (true) {
+		int cred_status = nrf_cloud_credentials_configured_check();
+
+		if (cred_status == -ENOTSUP) {
+			/* One or more necessary credentials are missing, try again later. */
+			k_sleep(K_SECONDS(20));
```

Comment:
I'm not in love with this solution, but I also don't see an event-based way to do it.

The sleep time should probably be configurable.

Reply:
Ah, yeah, that is probably the better way of handling this, at least in the short term. I'll switch to that.